RestSetAcls/RestSetAcls.psm1

```diff
@@ -0,0 +1,248 @@
+function Ask([Parameter(Mandatory=$true)][string] $question)
+{
+    while ($true) {
+        $yn = Read-Host "${question} [Y/n]"
+        $yn = $yn.Trim().ToLower()
+        if ($yn -eq 'n') {
+            return $false
+        } elseif ($yn -eq '' -or $yn -eq 'y') {
+            return $true
+        }
+        Write-Host "Invalid answer '$yn'. Answer with either 'y' or 'n'" -ForegroundColor Red
+    }
+}
+
+function Get-SpecialCharactersPrintable {
+    # Windows Terminal supports it
+    if ($env:WT_SESSION) {
+        return $true
+    }
+
+    # PowerShell 6+ supports it
+    if ($PSVersionTable.PSVersion.Major -ge 6) {
+        return $true
+    }
+
+    # Older versions don't
+    return $false
+}
+
+function Get-AzureFilesRecursive {
+    param (
+        [Parameter(Mandatory=$true, ValueFromPipeline=$true, ValueFromPipelineByPropertyName=$true)]
+        [Object[]]$Directory,
```

Comment:
This is a good point, thank you! I changed from `Object[]` to `AzureStorageBase[]`. In the initial call, I made the implicit conversion of `AzureStorageFileDirectory` to `AzureStorageFileDirectory[]` explicit. 

It needs to be an array because the `Get-AzStorageFile -Directory $file.CloudFileDirectory` call actually returns `Object[]`.